CIP-DappConnector/CIP-dapp-connector.md

```diff
@@ -237,6 +243,27 @@ Errors: `APIError`, `TxSendError`
 
 As wallets should already have this ability, we allow dApps to request that a transaction be sent through it. If the wallet accepts the transaction and tries to send it, it shall return the transaction id for the dApp to track. The wallet is free to return the `TxSendError` with code `Refused` if they do not wish to send it, or `Failure` if there was an error in sending it (e.g. preliminary checks failed on signatures).
 
+## Events
+
+In addition to the API methods that are actively called, the connector also must emit the following events. All methods events are required to be implemented.
+
+TODO: event emission method? Possible methods:
+
+1) Emitted event via `window.addEventListener(eventName , e => void)`
+2) Emitted message via `window.postMessage({eventName: string}, ...)`
+3) Some kind of callback registration i.e. `wallet.onDisconnect(() => void)` or `wallet.onEvent(eventName => void)`
+4) A combination of the two (event/message but with callback on `wallet` object as well
+5) Other?
+
+### wallet_disconnected
+
+Emitted when the user disconnects (not just changes) their current account from the page. This means that all `api` methods will return with an `APIError.Refused` error and a new `api` object must be obtained from `wallet.enable()` to continue to interact with the user's wallet.
+
+### account_changed
+
+Emitted when the user changes accounts (i.e. different root key pair and/or network id). The same `api` object will continue to work but will now return results based on the new account. After this event is received dApps should check `api.getNetworkId()` as changing accounts can also change the network.
```

Comment:
I'm not entirely sure this is good to have the old `api` object continue to work since dApps could mix results between accounts accidentally until they properly react to the event being dispatched. However if we have all `api` methods return something like a new `APIError.AccountChanged` error until the dApp recognizes the account changed, it it might be bad having them call `wallet.enable()` to get a new `api` object for the new account since `api.enable()` is documented as asking for the user's permission whereas this would be reacting to the user's action in the first place. We could have this event include the new `api` instance as a parameter but I do like the idea of having all params/returns for this API being JSON-serializable in case we extend it to another method beyond code injection in the future and that would violate that for events. Does anyone have any suggestions/comments concerning this?